test/eventinge2erekt/features/encryption.go

```diff
@@ -59,7 +64,8 @@ func VerifyEncryptedTrafficToActivator(refs []corev1.ObjectReference, since time
 			PodLogOptions: &corev1.PodLogOptions{Container: "istio-proxy", SinceTime: &metav1.Time{Time: since}},
 			JSONLogFilter: func(m map[string]interface{}) bool {
 				return GetMapValueAsString(m, "path") == "/" &&
-					GetMapValueAsString(m, "authority") == privateURL.Host
+					GetMapValueAsString(m, "authority") == privateURL.Host &&
+					GetMapValueAsString(m, "response_code") == responseCode
```

Comment:
If we want to make sure the traffic is blocked, we should perhaps not just check "it contains a line with a 403 response", but also for "it never contains a 2xx response"

Reply:
Done.